### PR TITLE
Feat/events

### PR DESCRIPTION
**Pull request by Raajheer1**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -4,16 +4,18 @@ import piniaPluginPersistedState from "pinia-plugin-persistedstate";
...
 import "flowbite";
+import "v-calendar/dist/style.css";
 
 const app = createApp(App);
 
 app.component("Alert", Alert);
 app.component("AuthLink", AuthLink);
 
-app.use(router).use(createPinia().use(piniaPluginPersistedState)).mount("#app");
+app.use(router).use(createPinia().use(piniaPluginPersistedState)).use(VCalendar, {}).mount("#app");
```
Comment (dhawton) [inline src/main.ts:21]: Don't set this up globally.. that creates unneeded bloat. Load it as a component where needed instead. https://vcalendar.io/vue-3.html#method-3-use-components-as-needed

File: src/views/pages/EventInfo.vue [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+const deleteEvent = async (): Promise<void> => {
+  if (canEditEvent()) {
+    try {
+      const result = await ZDVAPI.delete(`/v1/events/${id}`);
+      if (result.status === 204) {
+        await eventStore.fetchEvents();
+        await router.push(`/events/`);
+      }
+    } catch (error) {
+      console.error(error);
```
Comment (dhawton) [inline src/views/pages/EventInfo.vue:221]: Something should happen here so the user knows the delete failed. Look at the Alert component.

File: src/views/pages/EventsPage.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import EventCard from "@/views/partials/events/EventCard.vue";
+import useEventStore from "@/stores/event";
+import { useRouter } from "vue-router";
+
+const router = useRouter();
+
+const fetched = ref(false);
+const error = ref("");
+const eventStore = useEventStore();
+// const { events } = storeToRefs(eventStore);
```
Comment (dhawton) [inline src/views/pages/EventsPage.vue:30]: Remove this if not needed.

File: src/views/partials/events/SignupModal.vue [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        >
+          <svg
+            aria-hidden="true"
+            class="w-5 h-5"
+            fill="currentColor"
+            viewBox="0 0 20 20"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path
+              fill-rule="evenodd"
```
Comment (dhawton) [inline src/views/partials/events/SignupModal.vue:23]: Might be cleaner to use font awesome as it's already loaded?

File: src/views/partials/events/CreateEventModal.vue [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  <div v-show="isOpen" class="absolute inset-0 flex items-center justify-center bg-black-deep bg-opacity-70">
+    <div class="relative bg-white rounded-lg shadow dark:bg-black-light w-1/2">
+      <div class="flex justify-between items-start p-4 rounded-t border-b dark:border-gray-600">
+        <h3 class="text-xl font-semibold text-gray-900 dark:text-white m-0">Create Event Position</h3>
+        <button
+          type="button"
+          class="text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm p-1.5 ml-auto inline-flex items-center dark:hover:bg-gray-600 dark:hover:text-white"
+          @click="toggleModal()"
+        >
+          <svg
```
Comment (dhawton) [inline src/views/partials/events/CreateEventModal.vue:19]: Might be cleaner to use font awesome as it's already loaded?

File: src/main.ts (hunk 1 above)
Comment (Raajheer1) [reply src/main.ts]: Fixed in [6a787f1]

File: src/views/pages/EventInfo.vue (hunk 2 above)
Comment (Raajheer1) [reply src/views/pages/EventInfo.vue]: Fixed in [b401cbc]

File: src/views/pages/EventsPage.vue (hunk 3 above)
Comment (Raajheer1) [reply src/views/pages/EventsPage.vue]: Fixed in [b401cbc]

File: src/views/partials/events/SignupModal.vue (hunk 4 above)
Comment (Raajheer1) [reply src/views/partials/events/SignupModal.vue]: Fixed in [b401cbc]

File: src/views/partials/events/CreateEventModal.vue (hunk 5 above)
Comment (Raajheer1) [reply src/views/partials/events/CreateEventModal.vue]: Fixed in [b401cbc]

**Review by dhawton — APPROVED** [review]

**Pull request merged**